SchedulePlatform.Service/ServiceProvidedService.cs

```diff
@@ -1,55 +1,120 @@
+            var allServices = _serviceRepository.GetAll();
+
+            return _mapper.Map<IEnumerable<ServiceProvidedResponseModel>>(allServices);
         }
 
-        public ServiceProvided Add(ServiceProvided serviceP)
+        public ServiceProvidedResponseModel Add(ServiceProvidedRequestModel serviceP)
         {
-            var serviceToAdd = new ServiceProvided
-            {
-                Id = Guid.NewGuid(),
-                Description = serviceP.Description,
-                NameServiceProvided = serviceP.NameServiceProvided,
-                Price = serviceP.Price,
-                Type = serviceP.Type,
-                UrlPicture = serviceP.UrlPicture
-            };
+            var serviceToAdd = _mapper.Map<ServiceProvided>(serviceP);
 
-            return _serviceRepository.Add(serviceToAdd);
+            _serviceRepository.Add(serviceToAdd);
+
+            return _mapper.Map<ServiceProvidedResponseModel>(serviceToAdd);
         }
 
-        public ServiceProvided? GetById(Guid id)
+        public ServiceProvidedResponseModel GetById(Guid id)
         {
-            return _serviceRepository.GetById(id);
+            var findService = _serviceRepository.GetById(id);
+
+            return _mapper.Map<ServiceProvidedResponseModel>(findService);
         }
 
-        public ServiceProvided Update(ServiceProvided serviceP)
+        public IEnumerable<ServiceProvidedResponseModel> GetAllServicesByNutritionistId(Guid nutritionistId)
         {
-            return _serviceRepository.Update(serviceP);
+            if (_nutritionistServiceRepository.GetAll().ToList().Find(n=>n.NutritionistId==nutritionistId)==null)
+            {
+                throw new Exception("The nutrionist has no related services");
+            }
+
+            var fullList = _nutritionistServiceRepository.GetAll().ToList();
+            var allServices = _nutritionistServiceRepository.GetAll().ToList().Where(n => n.NutritionistId == nutritionistId);
+            var tempList = allServices.Select(l => l.ServiceId).ToList();
+
+            var tempListServices = new List<NutritionistService>();
+
+            foreach (var item in fullList)
+            {
+                foreach (var id in tempList)
+                {
+                    if (item.ServiceId==id)
+                    {
+                        tempListServices.Add(item);
+                    }
+                }
+            }
+
+            return _mapper.Map<IEnumerable<ServiceProvidedResponseModel>>(tempListServices);
         }
 
-        public ServiceProvided Delete(Guid id, ServiceProvided serviceP)
+        public UpdateServiceProvidedResponseModel Update(Guid id, UpdateServiceProvidedRequestModel serviceP)
         {
-            return _serviceRepository.Delete(id, serviceP);
+            var findService = _serviceRepository.GetById(id);
+
+            if (!string.IsNullOrEmpty(serviceP.Description))
+            {
+                findService.Description = serviceP.Description;
```

Comment:
Null check is mandatory on findService otherwise you will get null exception on line 80 if the service does not exist in the database.